### PR TITLE
11.0 mig auth session timeout

### PR DESCRIPTION
**Pull request by nadiaafa**

### CODE REVIEW

**Comment by ddufresne** [comment]
@nadiaafa :+1: tested the module. Seems to work fine.

However, as mentioned by @pedrobaeza, the module should be migrated to https://github.com/OCA/server-auth/pulls.

**Comment by nadiaafa** [comment]
@ddufresne thank you for the review, i will move it to https://github.com/OCA/server-auth/pulls

**Comment by ddufresne** [comment]
@nadiaafa excellent, looking forward reviewing under https://github.com/OCA/server-auth

File: auth_session_timeout/models/res_users.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                utime(path, None)
+            except OSError as e:
+                _logger.exception(
+                    'Exception updating session file access/modified times.',
+                )
+
+    @classmethod
+    def check(cls, *args, **kwargs):
+        res = super(ResUsers, cls).check(*args, **kwargs)
+        http.request.env.user._auth_timeout_check()
```
Comment (christophlsa) [inline auth_session_timeout/models/res_users.py:108]: @nadiaafa If you do a RPC request then this fails with this error (see LRMIG-679 intern):

```
   File "/opt/odoo/odoo/service/wsgi_server.py", line 124, in wsgi_xmlrpc
     result = odoo.http.dispatch_rpc(service, method, params)
   File "/opt/odoo/odoo/http.py", line 115, in dispatch_rpc
     result = dispatch(method, params)
   File "/opt/odoo/odoo/service/model.py", line 35, in dispatch
     security.check(db,uid,passwd)
   File "/opt/odoo/odoo/service/security.py", line 13, in check
     return res_users.check(db, uid, passwd)
   File "/opt/odoo/addons_extern/auth_session_timeout/models/res_users.py", line 108, in check
     http.request.env.user._auth_timeout_check()
   File "/usr/local/lib/python3.5/dist-packages/werkzeug/local.py", line 343, in __getattr__
    return getattr(self._get_current_object(), name)
   File "/usr/local/lib/python3.5/dist-packages/werkzeug/local.py", line 302, in _get_current_object
     return self.__local()
   File "/usr/local/lib/python3.5/dist-packages/werkzeug/local.py", line 135, in _lookup
     raise RuntimeError('object unbound')
 RuntimeError: object unbound
```

I fixed this in our project with replacing:

```python
    @classmethod
    def check(cls, *args, **kwargs):
        res = super(ResUsers, cls).check(*args, **kwargs)
        if http and http.request and http.request.env and \
                http.request.env.user:
            http.request.env.user._auth_timeout_check()
        return res
```

**Comment by ddufresne** [comment]
@nadiaafa any update on this migration ? Are you going to propose a PR on https://github.com/OCA/server-auth ?

**Comment by Change2improve** [comment]
Any news with this PR or if there will be a new one in https://github.com/OCA/server-auth ?

**Comment by pedrobaeza** [comment]
No news, so feel free to start a new PR there.

**Comment by zaoral** [comment]
@pedrobaeza 
I have created a new PR https://github.com/OCA/server-auth/pull/32 that took this PR changes and apply some changes to follow OCA guideline

**Pull request closed without merging**